service-base/src/main/java/org/eclipse/hono/service/quarkus/AbstractServiceApplication.java

```diff
@@ -94,38 +96,14 @@ protected void logJvmDetails() {
         }
     }
 
-    /**
-     * Sets common metrics tags.
-     * <p>
-     * This default implementation does nothing. Subclasses should override this method to set the metrics tags,
-     * if not already set elsewhere.
-     */
-    protected void setCommonMetricsTags() {
-        // nothing done by default
-    }
-
-    /**
-     * Enables collection of JVM related metrics.
-     * <p>
-     * Enables collection of Memory, Thread, GC and Processor metrics.
-     */
-    protected void enableJvmMetrics() {
-        new ProcessorMetrics().bindTo(meterRegistry);
-        new JvmMemoryMetrics().bindTo(meterRegistry);
-        new JvmThreadMetrics().bindTo(meterRegistry);
-        new FileDescriptorMetrics().bindTo(meterRegistry);
-        this.jvmGcMetrics = new JvmGcMetrics();
-        jvmGcMetrics.bindTo(meterRegistry);
-    }
-
     /**
      * Registers additional health checks.
      *
      * @param provider The provider of the health checks to be registered (may be {@code null}).
      */
     protected final void registerHealthchecks(final HealthCheckProvider provider) {
         Optional.ofNullable(provider).ifPresent(p -> {
-            LOG.debug("registering health checks [provider: {}]", p.getClass().getName());
+            LOG.debug("registering legacy health checks [provider: {}]", p.getClass().getName());
```

Comment:
What is meant by `legacy` here?
I would have expected the `registerHealthCheckProvider` method to be marked as deprecated then.

Reply:
I wanted to express that the existing checks are not implemented based on the MicroProfile Health spec. Marking the method as deprecated might be a good idea as well.

Reply:
IMHO we will not have the time to migrate and doing it later shouldn't be a problem because this does not affect the external APIs I will then mark the method as `@Deprecated` ...